includes/functions.php

```diff
@@ -16,42 +16,71 @@
  * Register a new batch process.
  *
  * @param  array $args Arguments for the batch process.
- * @throws Exception Only post & user are accepted $args['type'].
+ * @throws Exception If the batch processor doesn't extend the Batch abstract class.
  */
 function register_batch_process( $args ) {
 	if ( empty( $args['type'] ) ) {
 		$args['type'] = '';
 	}
 
-	switch ( $args['type'] ) {
+	/**
+	 * Filter the batch processor to be used with the batch process being registered.
+	 *
+	 * @param Batch  $batch_processor The batch processor to use, defaults to null.
+	 * @param string $type            Type of data for this batch process.
+	 * @param array  $args            Arguments for the batch process.
+	 */
+	$batch_processor = apply_filters( 'loco_register_batch_process_processor', null, $args['type'], $args );
```

Comment:
It looks like you might be looking at one of the first commits for this PR. That's set differently now: https://github.com/reaktivstudios/locomotive/pull/119/files#diff-f81f3dff36bf04b7a070d642c858e883